docs/tunnel.md

```diff
@@ -0,0 +1,56 @@
+# Minikube Tunnel Design Doc
```

Comment:
Did you mean to include this design doc in this PR?

Reply:
Nope, thanks!